lib/active_record_doctor/detectors/missing_unique_indexes.rb

```diff
@@ -76,16 +89,8 @@ def has_ones_without_indexes # rubocop:disable Naming/PredicateName
         end
       end
 
-      def supported_validator?(validator)
-        validator.options[:if].nil? &&
-          validator.options[:unless].nil? &&
-          validator.options[:conditions].nil? &&
-
-          # In Rails 6, default option values are no longer explicitly set on
-          # options so if the key is absent we must fetch the default value
-          # ourselves. case_sensitive is the default in 4.2+ so it's safe to
-          # put true literally.
-          validator.options.fetch(:case_sensitive, true)
+      def conditional_validator?(validator)
+        (validator.options.keys & [:if, :unless, :conditions]).any?
```

Comment:
I think `present?` would be slightly more appropriate here, as our intent is to test whether the intersection is non-empty, not whether it contains a truthy value.